README.md

```diff
@@ -339,8 +339,61 @@ LIT_SERVER_API_KEY=supersecretkey python main.py
 Clients are expected to auth with the same API key set in the `X-API-Key` HTTP header.
 
 </details>
+
+
+<details>
+  <summary>Stream long responses</summary>
+
 &nbsp;
 
+`LitServer` can stream longer responses, such as LLM generated text.
+
+To enable streaming, you need to implement `LitAPI.predict` and `LitAPI.encode_response` as a generator (a Python
+function that yields output) and set `LitServer(..., stream=True)`.
+
+For example, streaming long responses generated over time:
+
+```python
+from typing import Generator
+
+from litserve.api import LitAPI
+import json
+
+from litserve import LitServer
+from pydantic import BaseModel
+
+
+class Request(BaseModel):
+    number: int
+
+
+class SimpleStreamAPI(LitAPI):
+    def setup(self, device) -> None:
+        self.num_outputs = 100
+
+    def decode_request(self, request: Request) -> str:
+        return request.number
+
+    def predict(self, x) -> Generator:
+        for i in range(self.num_outputs):
+            yield x**i
+
+    def encode_response(self, output: Generator) -> Generator:
+        for out in output:
+            yield json.dumps({"output": out})
+
+
+if __name__ == "__main__":
+    api = SimpleStreamAPI()
+    server = LitServer(api, stream=True, timeout=30)
```

Comment:
why are you adding the timeout? is it necessary for straming?
again... don't add unnecessary concepts